packages/contracts/contracts/veION/veION.sol

```diff
@@ -133,42 +127,39 @@ contract veION is Ownable2StepUpgradeable, ERC721Upgradeable, IveION {
     if (unlockTime <= oldLocked.end) revert LockDurationNotInFuture();
     if (unlockTime > block.timestamp + MAXTIME) revert LockDurationTooLong();
 
-    _depositFor(_tokenAddress, _tokenId, 0, unlockTime, false, oldLocked, DepositType.INCREASE_UNLOCK_TIME, _lpType);
+    _depositFor(_lpType, _tokenId, 0, unlockTime, false, oldLocked, DepositType.INCREASE_UNLOCK_TIME);
   }
 
   function lockAdditionalAsset(
-    address _tokenAddress,
+    LpTokenType _lpType,
     uint256 _tokenAmount,
     uint256 _tokenId,
     uint256 _duration,
     bool _stakeUnderlying
   ) external {
     if (_tokenAmount == 0) revert ZeroAmount();
     if (!_isApprovedOrOwner(_msgSender(), _tokenId)) revert NotApprovedOrOwner();
-    if (!s_whitelistedToken[_tokenAddress]) revert TokenNotWhitelisted();
 
-    LpTokenType lpType = s_lpType[_tokenAddress];
-    LockedBalance storage lockedBalance = s_locked[_tokenId][lpType];
-    s_assetsLocked[_tokenId].add(_tokenAddress);
+    LockedBalance storage lockedBalance = s_locked[_tokenId][_lpType];
+    s_assetsLocked[_tokenId].add(uint256(_lpType));
```

Comment:
won't this line convert from an ENUM to a uint256 when adding?  If it is `Mode_Verlodrome_5050_ION_MODE` wouldn't that be 0 since it is the first ENUM option?  I think you need to test this to confirm it does what you want it to do.